receiver/kymastatsreceiver/internal/modulediscovery/client_test.go

```diff
@@ -0,0 +1,53 @@
+package modulediscovery
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	discoveryfake "k8s.io/client-go/discovery/fake"
+	clienttesting "k8s.io/client-go/testing"
+)
+
+func TestDiscover(t *testing.T) {
+	discovery := discoveryfake.FakeDiscovery{
+		Fake: &clienttesting.Fake{
+			Resources: []*metav1.APIResourceList{
+				{
+					GroupVersion: "operator.kyma-project.io/v1beta1",
```

Comment:
what is the expected behaviour in case a resource was available in v1alpha1 but never got a v1beta1?
Can you please add a test?

Reply:
I checked the code, and it's a problem indeed; taking a preferred version is not the right solution. I will think how we can solve it.